subsys/bluetooth/services/hids.c

```diff
@@ -477,16 +477,21 @@ static void hids_input_report_ccc_changed(struct bt_gatt_attr const *attr,
 	    CONTAINER_OF((struct _bt_gatt_ccc *)attr->user_data,
 			 struct bt_hids_inp_rep, ccc);
 
+	uint8_t report_id = inp_rep->id;
+	enum bt_hids_notify_evt evt;
+
 	if (value == BT_GATT_CCC_NOTIFY) {
 		LOG_DBG("Notification has been turned on");
-		if (inp_rep->handler != NULL) {
-			inp_rep->handler(BT_HIDS_CCCD_EVT_NOTIFY_ENABLED);
-		}
+		evt = BT_HIDS_CCCD_EVT_NOTIFY_ENABLED;
 	} else {
 		LOG_DBG("Notification has been turned off");
-		if (inp_rep->handler != NULL) {
-			inp_rep->handler(BT_HIDS_CCCD_EVT_NOTIFY_DISABLED);
-		}
+		evt = BT_HIDS_CCCD_EVT_NOTIFY_DISABLED;
+	}
+
+	if (inp_rep->handler_ext != NULL) {
+		inp_rep->handler_ext(report_id, evt);
```

Comment:
Similar update seems needed for `hids_boot_mouse_inp_rep_ccc_changed` and `hids_boot_kb_inp_rep_ccc_changed`. Alternatively we could assert or return error to prevent user from registering the extended callback there (report ID parameter is not used for boot reports).

Reply:
Discussed offline. Overlooked that HID boot reports use another structure type. We can leave it as is.